app/controllers/articles.controllers.js

```diff
@@ -11,8 +11,17 @@ exports.getArticleById = (req, res, next) => {
 }
 
 exports.getArticles = (req, res, next) => {
-    selectArticles()
-    .then((articles) => res.status(200).send({articles}))
+    const {topic} = req.query
+    const articlesQuery = selectArticles(topic)
+    const queries = [articlesQuery]
+    if(topic){
+        const topicExists = checkTopicExists(topic)
+        queries.push(topicExists)
+    }
+    Promise.all(queries)
+    .then((results) => {
+        const articles = results[0]
+        res.status(200).send({articles})})
```

Comment:
This is a really nice way to go about checking the topic exists in the database when this query is used 👍